### ISSUE TITLE
Is there a way to tell fansly downloader ng to download a plaintext copy of all the text in all the posts on a timeline for a model I am subscribed to?

### ISSUE THREAD

**Issue by left1000** [prof79/fansly-downloader-ng#25]
Is there a way to tell fansly downloader ng to download a plaintext copy of all the text in all the posts on a timeline for a model I am subscribed to?

I'm a completionist afterall.... it'd be nice to be able to read what they wrote when posting each of the 100s of videos, if I wanted to.

**Comment by prof79** [comment]
Good point @left1000 . I see that each post in a "timeline" request also yields the textual content.

Question is - "all the text in all the posts" - would it suffice to have a folder `Text` with all `<post_id>.txt` in it?
Or regarding "what they wrote when posting" - do you expect a correlation between text and media items?

First would be fairly easy to do though requiring some care introducing the new folder structure. Second would be much more challenging how to properly relate those also from a design perspective.

**Comment by PrincessRTFM** [comment]
It would require a new option _and_ new code to handle it, but you could separate posts into their own folders named by the post date/time, containing a text file (does fansly allow formatting in posts? maybe a markdown file or even simple HTML) and whatever media the post had. The resulting file structure, if this option was enabled, would then be something like

- `/` (the username whose content was downloaded for `Local_directory` mode, or target folder if one is specified)
  - `$post_source` ("Timeline", "Messages", whatever)
    - `$timestamp` (based on ISO8601 for sorting, maybe `YYYY-MM-DD HH-mm-ss` or similar)
      - `content.txt`
      - `$media_id.jpg`
      - `$media_id.mp3`
      - `$media_id.m4a`
      - etc (whatever other media was present)

Personally, I'd prefer this kind of structure even without the post text just so I can view the downloaded content in chronological order.

All that said, I don't know how hard this would be to implement, but I know it would certainly be a significant effort.

**Comment by left1000** [comment]
Really any implementation is fine. I'm just so spoiled by gallery-dl. But as long as the text file exists, well, I could hopefully use date/time/order to match them up.

The way gallery-dl does it is by using post_id in the filename of the videos and post_id in the filename of the text files, the text files go on my set up in a subfolder to avoid clutter.
